Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -119,6 +118,17 @@ private void CleanExpiredSessions()
             }
         }
 
+        internal void ValidateSecurePassword(SecureString password)
+        {
+            if (!ExtractPassword(Password).Equals(ExtractPassword(password)))
+            {
+                throw new Exception("Could not get a pool because of password mismatch");
+            }
+        }
+
+        private string ExtractPassword(SecureString password) =>
```

Comment:
minor - this could have been part of SecureStringHelper class

Reply:
this converts null to empty string for easy comparison. This is not what we should expect from the general purpose class like SecureStringHelper, so I would rather leave it as is. We could discuss it.